api.v2/java.v2/src/main/java/com/listeners/TestNGListenerForPractiTest.java

```diff
@@ -0,0 +1,75 @@
+package com.listeners;
+
+import com.Config.GeneralConfig;
+import com.practitest.api.example.PractiTestWriter;
+import com.practitest.integration.ExtractTests;
+import com.webdriver.example.Utils.Log;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+import java.util.List;
+
+public class TestNGListenerForPractiTest implements ITestListener {
+
+        protected String setID = null;
+
+
+        @Override
+        public void onTestStart(ITestResult result) {
+        }
+
+        @Override
+        public void onTestSuccess(ITestResult result) {
+                List<String> instanceID = PractiTestWriter.getInstancesByTestIDAndTestSetID(result.getMethod().getDescription(), this.setID);
+                PractiTestWriter.submitResults(instanceID.get(0), 0);
+        }
+
+        @Override
+        public void onTestFailure(ITestResult result) {
+                List<String> instanceID = PractiTestWriter.getInstancesByTestIDAndTestSetID(result.getMethod().getDescription(), this.setID);
+                PractiTestWriter.submitResults(instanceID.get(0), 1);
```

Comment:
We need to send more information (how it failed, what was the exception) than just status code to PractiTest.

Reply:
Added throwable message to test result